app/controllers/PrepareUploadController.scala

```diff
@@ -49,13 +50,29 @@ class PrepareUploadController @Inject()(
       onlyAllowedServices[JsValue] { (_, consumingService) =>
 
         withJsonBody[UploadSettings] { (fileUploadDetails: UploadSettings) =>
-          Logger.debug(s"Processing request: [$fileUploadDetails].")
-          val sessionId = hc(request).sessionId.map(_.value).getOrElse("n/a")
-          val requestId = hc(request).requestId.map(_.value).getOrElse("n/a")
-          val result: PreparedUpload =
-            prepareUploadService.prepareUpload(fileUploadDetails, consumingService, requestId, sessionId)
-          Future.successful(Ok(Json.toJson(result)))
+
+          withAllowedCallbackProtocol(fileUploadDetails.callbackUrl) {
+            Logger.debug(s"Processing request: [$fileUploadDetails].")
+
+            val sessionId = hc(request).sessionId.map(_.value).getOrElse("n/a")
+            val requestId = hc(request).requestId.map(_.value).getOrElse("n/a")
+            val result: PreparedUpload =
+              prepareUploadService.prepareUpload(fileUploadDetails, consumingService, requestId, sessionId)
+
+            Future.successful(Ok(Json.toJson(result)))
+          }
         }
       }
     }
+
+  private[controllers] def withAllowedCallbackProtocol[A](callbackUrl: String)
+                                            (block: => Future[Result]): Future[Result]= {
+    if (callbackUrl.startsWith("https")) {
```

Comment:
We don't validate if it's valid URL.
It's not necessarily very big issue but maybe worth taking a look.